core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -1996,6 +2059,104 @@ impl<R: WriteableRocks> StateManagerDatabase<R> {
 
         last_state_version
     }
+
+    fn batch_update_entity_listing_indices(
+        &self,
+        db_context: &TypedDbContext<R, BufferedWriteSupport<R>>,
+        state_version: StateVersion,
+        substate_changes: &BySubstate<SubstateChangeAction>,
+    ) {
+        for (index_within_txn, node_id) in substate_changes.iter_node_ids().enumerate() {
+            let type_info_creation = substate_changes.get(
+                node_id,
+                &TYPE_INFO_FIELD_PARTITION,
+                &TypeInfoField::TypeInfo.into(),
+            );
+            let Some(type_info_creation) = type_info_creation else {
+                continue;
+            };
+            let SubstateChangeAction::Create { new } = type_info_creation else {
+                panic!(
+                    "type info substate should be immutable: {:?}",
```

Comment:
This seems quite scary. e.g. this won't be true if the blueprint version is updated eventually.

But the entity id and blueprint name should be fixed by the system, so I think it's fine to just ignore updates.